docs/book/src/tasks/certs/auto-rotate-certificates-in-kcp.md

```diff
@@ -0,0 +1,59 @@
+## Automatically rotating certificates using Kubeadm Control Plane provider
+
+When using Kubeadm Control Plane provider (KCP) it is possible to configure automatic certificate rotations. KCP does this by triggering a rollout when the certificates on the machines are about to expire.
+
+### Configuring Machine Rollout
+
+To configure a rollout on the KCP machines you need to set `.rolloutBefore.certificatesExpiryDays` (minimum of 7 days).  
+
+Example: 
+```yaml
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: KubeadmControlPlane
+metadata:
+  name: example-control-plane
+spec:
+  rolloutBefore:
+    certificatesExpiryDays: 21 # trigger a rollout if certificates expire within 21 days
+  kubeadmConfigSpec:
+    clusterConfiguration:
+      ...
+    initConfiguration:
+      ...
+    joinConfiguration:
+      ...
+  machineTemplate:
+    infrastructureRef:
+      ...
+  replicas: 1
+  version: v1.23.3
+``` 
+
+TODO: Add a few lines on recommendations to set the `certificatesExpiryDays` to a big value so that all the machines will have time to completely rollout.
+
+### Calculating Certificate Expiry
+
+KCP uses the value in the corresponding Control Plane machine's `Machine.Status.CertificatesExpiryDate` to check if a machine's certificates are going to expire and if it needs to be rolled out.  
+
+`Machine.Status.CertificatesExpiryDate` get its value from one of the following 2 places:
+
+* `machine.cluster.x-k8s.io/certificates-expiry` annotation value on the Machine object
```

Comment:
```suggestion
* `machine.cluster.x-k8s.io/certificates-expiry` annotation value on the Machine object; this annotation is not applied by default and it can be set by users to opt-in in this feature for existing control plane machines (see instructions below)
```